files/default/scripts/tests/berkshelf.sh

```diff
@@ -1,6 +1,18 @@
 #!/bin/bash
+
+BERKSHELF_TESTS_DIR=$PROXY_TESTS_DIR/tests/berkshelf
+
 # Clear the berkshelf
 run_cmd rm -rf ~/.berkshelf
 
-# Install apache2
-run_cmd timeout -k 60 -s 9 60 berks install -b $PROXY_TESTS_DIR/tests/berkshelf/Berksfile || PROXY_TEST_RESULT=failed
+# Attempt a berks install using the sample Berksfile
+run_cmd berks install --berksfile=$BERKSHELF_TESTS_DIR/Berksfile || PROXY_TEST_RESULT=failed
+
+# Attempt to do a berks upload to a Chef server.
+#
+# This requires doing a berks install to get everything resolved.
+#
+# We don't need to authenticate successfully, just check to see if we got a 401
+# response. If we get this response it means we connected successfully through
+# the proxy.
+run_cmd berks install --berksfile=$BERKSHELF_TESTS_DIR/Berksfile && berks upload --berksfile=$BERKSHELF_TESTS_DIR/Berksfile | grep HTTPUnauthorized || PROXY_TEST_RESULT=failed
```

Comment:
I got it set up to use the client-creator user that's used elsewhere, but that user doesn't have update permissions.

Not sure if we want something in a public repo to contain credentials that allow cookbook upload anyway. Do we want to try making that work? For completeness I'd say let's make it actually upload something, but for safety I'd say that getting a 401 is sufficient to be able to determine if the proxy settings are working.